junit-jupiter-engine/src/test/kotlin/org/junit/jupiter/api/ScopedContext.kt

```diff
@@ -0,0 +1,14 @@
+/*
+ * Copyright 2015-2020 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * https://www.eclipse.org/legal/epl-v20.html
+ */
+package org.junit.jupiter.api
+
+fun <T> contextScope(block: ContextScope.() -> T): T = object : ContextScope {
```

Comment:
The idea was that `apply` can only be called from `ContextScope` so that when calling it from within `assertThrows` it verifies that the scope has been inlined, thus behaving in the same was as `CoroutineScope.launch { }`